.gitlab/common/macos.yml

```
@@ -16,6 +16,24 @@
       dda inv -- -e macos.remove-inactive-versions -l python -t "$PYTHON_VERSION" || true
       dda inv -- -e macos.remove-inactive-versions -l go -t "$(cat .go-version)" || true
     fi
+  # Create custom temporary folder to isolate jobs from each other
+  # We have to symlink it to /tmp/gitlabci to avoid some path length issues (sockets should be <= 104 characters on MacOS)
+  - |
+    if [ -z "$TMPDIR" ]; then
+      echo "TMPDIR must be set" >& 2
+      exit 1
+    fi
+    # Remove the trailing slashes etc.
+    export OLDTMPDIR="$(readlink -m "$TMPDIR")"
+    sudo rm -rf "$OLDTMPDIR/gitlabci"
+    NEWTMPDIR="$OLDTMPDIR/gitlabci/$CI_JOB_ID"
+    mkdir -p "$NEWTMPDIR"
+    sudo chmod 777 "$NEWTMPDIR"
```

no need to change anything here but, for completeness, `mkdir` supports creation flags such as:

```suggestion
    mkdir -m 777 -p "$NEWTMPDIR"
```